### ISSUE TITLE
Building wheels for package in subdirectory of project

### ISSUE THREAD

**Issue by filips123** [joerick/cibuildwheel#294]
Is there any way to build wheels for a package in a subdirectory of the project?

Example of this is when GitHub project contains some C++ code, while Python bindings are located in some subdirectory of that project, like `bindings/python`. Then it's also not possible to just set project directory to that subdirectory because the `BEFORE_BUILD` script would expect the main C++ code to be normally present, while they won't be copied because they would be outside project directory.

**Comment by Czaki** [comment]
Why using `../..` is not enough to solve this problem?

**Comment by filips123** [comment]
But where could you use `../..`?

**Comment by Czaki** [comment]
>But where could you use ../..?

In before build. But I see problem with docker. 

I'm not sure if I good understand your problem. If you would like to use `cibuildwheel` to build wheel with python binding to library. Then this library is python independent, Then using CIBW_BEFORE_BUILD is bad idea because it is execute for each selected python. 

For such case in my project I build c/c++ libs before I call cibuildwheel (for linux it is done with dockerfile and configured custom manylinux image)

If during build of library there is need to have access to proper python interpreter then I do not understand layout with hidden setup.py.

**Comment by filips123** [comment]
Solution for `BEFORE_BUILD` that is executed too many times is to create a file and check it as described as temporarily solution in #54.

Although I agree that building library before running `cibuildwheel` and creating custom `mwnylinux` image would work, I think that this (specially creating custom image just for this build) would be more complicated than just specifying path to `setup.py` and script that is started once per environment.

**Comment by Czaki** [comment]
Why? Dockerfile will be simple:

```Dockerfile
FROM quay.io/pypa/manylinux2010_x86_64
WORKDIR /opt/imagecodecs/build_utils

COPY library .
RUN make

ENV some_path_set
```

then task:
```
docker build -f Dockerfile -t custom_manylinux .
```

And set 
```yaml
CIBW_MANYLINUX_X86_64_IMAGE: custom_manylinux
```

**Comment by YannickJadoul** [comment]
Did we not also have `/host` as a mount point for the host machine's file system? That's easier than using a custom docker image. The only problem is that it's not allowed on CircleCI's docker, I think.

**Comment by joerick** [comment]
Our solution was released in v1.4.0!
